reporthandling/helpers/v1/listing.go

```diff
@@ -89,62 +98,63 @@ func (all *AllLists) Append(status apis.ScanningStatus, str ...string) {
 
 	switch status {
 	case apis.StatusPassed:
-		all.passed = append(all.passed, str...)
+		if all.passed == nil {
+			all.passed = sets.NewString(str...)
+		} else {
+			all.passed.Insert(str...)
+		}
 	case apis.StatusFailed:
-		all.failed = append(all.failed, str...)
+		if all.failed == nil {
+			all.failed = sets.NewString(str...)
+		} else {
+			all.failed.Insert(str...)
+		}
 	case apis.StatusExcluded:
-		all.excluded = append(all.excluded, str...)
+		if all.excluded == nil {
+			all.excluded = sets.NewString(str...)
+		} else {
+			all.excluded.Insert(str...)
+		}
 	case apis.StatusSkipped:
-		all.skipped = append(all.skipped, str...)
+		if all.skipped == nil {
+			all.skipped = sets.NewString(str...)
+		} else {
+			all.skipped.Insert(str...)
+		}
 	default:
-		all.other = append(all.other, str...)
+		if all.other == nil {
+			all.other = sets.NewString(str...)
+		} else {
+			all.other.Insert(str...)
+		}
 	}
 }
 
 // Update AllLists objects with
 func (all *AllLists) Update(all2 *AllLists) {
-	all.passed = append(all.passed, all2.passed...)
-	all.failed = append(all.failed, all2.failed...)
-	all.excluded = append(all.excluded, all2.excluded...)
-	all.skipped = append(all.skipped, all2.skipped...)
-	all.other = append(all.other, all2.other...)
-}
-
-// ToUnique - Call this function only when setting the List
-func (all *AllLists) toUniqueBase() {
-	// remove duplications from each resource list
-	all.failed = str.SliceStringToUnique(all.failed)
-	all.excluded = str.SliceStringToUnique(all.excluded)
-	all.passed = str.SliceStringToUnique(all.passed)
-	all.skipped = str.SliceStringToUnique(all.skipped)
-	all.other = str.SliceStringToUnique(all.other)
+	all.passed.Insert(all2.Passed()...)
+	all.failed.Insert(all2.Failed()...)
+	all.excluded.Insert(all2.Excluded()...)
+	all.skipped.Insert(all2.Skipped()...)
+	all.other.Insert(all2.Other()...)
 }
 
 // ToUnique - Call this function only when setting the List
 func (all *AllLists) ToUniqueControls() {
-	all.toUniqueBase()
-
 	// remove failed from excluded list
-	all.excluded = trimUnique(all.excluded, all.failed)
+	all.excluded = all.excluded.Difference(all.failed)
```

Comment:
Very elegant idiom indeed. However, the innards of the Set reveal that we allocate a new map.
https://github.com/kubernetes/apimachinery/blob/373a5f752d44989b9829888460844849878e1b6e/pkg/util/sets/set.go#L108

Reply:
yup, you can do better than that 💪 at least a quick benchmark shows indeed very little (if any) improvements over the current solution